powersimdata/input/helpers.py

```diff
@@ -120,3 +130,325 @@ def get_interconnect(idx):
+    """Take a plant-column data frame and decompose it into plant-column data frames
+    for each resource.
+
+    :param pandas.DataFrame df: data frame, columns are plant id in grid.
+    :param str/list/tuple/set resources: resource(s) to use for decomposition.
+    :param powersimdata.input.grid.Grid grid: Grid instance.
+    :return: (*dict*) -- keys are resources, values are plant-column data frames.
+    """
+    _check_data_frame(df, "PG")
+    plant_id = set(df.columns)
+    _check_plants_are_in_grid(plant_id, grid)
+    resources = _check_resources_are_in_grid_and_format(resources, grid)
+
+    df_resources = {
+        r: df[get_plant_id_for_resources(r, grid) & plant_id].sort_index(axis=1)
+        for r in resources
+    }
+    return df_resources
+
+
+def decompose_plant_data_frame_into_areas(df, areas, grid):
+    """Take a plant-column data frame and decompose it into plant-column data frames
+    for areas.
+
+    :param pandas.DataFrame df: data frame, columns are plant id in grid.
+    :param dict areas: areas to use for decomposition. Keys are area types
+        ('*loadzone*', '*state*', or '*interconnect*'), values are
+        str/list/tuple/set of areas.
+    :param powersimdata.input.grid.Grid grid: Grid instance.
+    :return: (*dict*) -- keys are areas, values are plant-column data frames.
+    """
+    _check_data_frame(df, "PG")
+    plant_id = set(df.columns)
+    _check_plants_are_in_grid(plant_id, grid)
+    areas = _check_areas_are_in_grid_and_format(areas, grid)
+
+    df_areas = {}
+    for k, v in areas.items():
+        if k == "interconnect":
+            for i in v:
+                name = "%s interconnect" % " - ".join(i.split("_"))
+                df_areas[name] = df[get_plant_id_in_interconnects(i, grid) & plant_id]
+        elif k == "state":
+            for s in v:
+                df_areas[s] = df[get_plant_id_in_states(s, grid) & plant_id]
+        elif k == "loadzone":
+            for l in v:
+                df_areas[l] = df[get_plant_id_in_loadzones(l, grid) & plant_id]
+
+    return df_areas
+
+
+def decompose_plant_data_frame_into_areas_and_resources(df, areas, resources, grid):
+    """Take a plant-column data frame and decompose it into plant-column data frames
+    for each resources-areas combinations.
+
+    :param pandas.DataFrame df: data frame, columns are plant id in grid.
+    :param dict areas: areas to use for decomposition. Keys are area types
+        ('*loadzone*', '*state*' or '*interconnect*'), values are
+        str/list/tuple/set of areas.
+    :param str/list/tuple/set resources: resource(s) to use for decomposition.
+    :param powersimdata.input.grid.Grid grid: Grid instance.
+    :return: (*dict*) -- keys are areas, values are dictionaries whose keys are
+        resources and values are data frames indexed by (datetime, plant) where plant
+        include only plants of matching type and located in area.
+    """
+    # areas_resources = defaultdict(dict)
```

Comment:
Looks like we can remove the commented out line

Reply:
Done, with some additional simplification of the function as well.